rest-parameters.md

```diff
@@ -0,0 +1,64 @@
+# ES6 Rest parameters
+
+Functions can take a rest parameters that act as a placeholder for a non-finite number of arguments.
+
+```javascript
+function score(user, ...marks) {
+  const marks_count = marks.length;
+  let total = 0;
+
+  // sum all the marks together
+  for (let mark of marks) {
+    total += mark;
+  }
+
+  const score = Math.round(total / marks_count);
+
+  return `John has a score of ${score}`;
+}
+
+console.log(score("John", 12, 8, 8, 12, 15));
+
+// "John has a score of 11"
+```
+
+Rest parameters automatically build an array, so given this function:
+
+```javascript
+function sentence(...words) {
+  return words.join(" ");
+}
+```
+
+The following code will work as expected:
+
+```javascript
+console.log(sentence("You", "are", "awesome"));
+
+// "You are awesome"
+```
+
+While this code will convert the array into its string representation:
+
+```javascript
+console.log(sentence(["You", "are", "awesome"]));
```

Comment:
Let's move this in with the above example the other executed code output, great example.

```js
console.log(sentence(["You", "are", "awesome"])); // You,are,awesome